packages/backend/src/repositories/product.ts

```diff
@@ -29,8 +32,35 @@ export class ProductRepository {
       where: {
         type,
         categoryId,
+        status: ProductStatus.ACTIVE,
+        price: {
+          gt: +priceGt,
+          lte: +priceLt,
+        },
+      },
+    });
+  }
+
+  public async getAllItemsLength(query: ProductQuery) {
```

Comment:
maybe it make sense to get count in a one transaction with data? 
like 
`const [posts, totalPosts] = await prisma.$transaction([
  prisma.post.findMany({ where: { title: { contains: 'prisma' } } }),
  prisma.post.count(),
])`